extractor.go

```diff
@@ -460,6 +548,44 @@ func mapValidator(v abci.ValidatorUpdate) (*codec.Validator, error) {
 	}, nil
 }
 
+func mapValidators(v []*types.Validator) ([]*codec.Validator, error) {
+	validators := []*codec.Validator{}
+	if len(v) > 0 {
+		for _, v := range v {
+			valv := v
+			val, err := mapValidator(valv)
+			if err != nil {
+				return nil, err
+			}
+			validators = append(validators, val)
+		}
+	}
+	return validators, nil
+}
+
+func mapValidator(v *types.Validator) (*codec.Validator, error) {
+	nPK := &codec.PublicKey{}
+
+	key := v.PubKey
+	switch key.Type() {
+	case "Ed25519":
```

Comment:
These vals are defined as constants somewhere in the tendermint library (afaik), so we can probably use those instead